docs/stream.md

```diff
@@ -10,6 +10,40 @@ Consuming can be done using `kstreams.Stream`. You only need to decorate a `coro
         docstring_section_style: table
         show_signature_annotations: false
 
+## Dependency Injection and typing
```

Comment:
This a good chunk that we probably should add to the release notes once we release